### PR TITLE
start cluster checks when cluster servelet starts

### PR DESCRIPTION
**Pull request by BelSasha**

### CODE REVIEW

**Comment by BelSasha** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/run-house/runhouse/1104?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#1104** <a href="https://app.graphite.dev/github/pr/run-house/runhouse/1104?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#1103** <a href="https://app.graphite.dev/github/pr/run-house/runhouse/1103?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @BelSasha and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: tests/test_resources/test_clusters/test_cluster.py [hunk 1]
```diff
@@ -755,29 +755,29 @@ def test_send_status_to_db(self, cluster):
 
     @pytest.mark.level("minimal")
     @pytest.mark.clustertest
-    def test_status_scheduler_basic_flow(self, cluster):
+    def test_status_scheduler_basic_flow(self, ondemand_aws_https_cluster_with_auth):
```
Comment (jlewitt1) [inline tests/test_resources/test_clusters/test_cluster.py:758]: any reason this is a specific cluster fixture?

File: tests/test_resources/test_clusters/test_cluster.py [hunk 2]
```diff
@@ -755,29 +755,29 @@ def test_send_status_to_db(self, cluster):
...
+        cluster = ondemand_aws_https_cluster_with_auth
+        cluster.restart_server()
```
Comment (jlewitt1) [inline tests/test_resources/test_clusters/test_cluster.py:760]: do we need this?

File: tests/test_resources/test_clusters/test_cluster.py [hunk 3]
```diff
@@ -755,29 +755,29 @@ def test_send_status_to_db(self, cluster):
...
 
         if not cluster.config().get("resource_subtype") == "OnDemandCluster":
             pytest.skip(
                 "This test checking pinging cluster status to den, this could be done only on OnDemand clusters."
             )
 
+        cluster_uri = rh.globals.rns_client.format_rns_address(cluster.rns_address)
```
Comment (jlewitt1) [inline tests/test_resources/test_clusters/test_cluster.py:767]: maybe im missing something but what's the core change here?

File: tests/test_resources/test_clusters/test_cluster.py (hunk 1 above)
Comment (BelSasha) [reply tests/test_resources/test_clusters/test_cluster.py:758]: nope, it was used for my local testing, reverted it to use a generic cluster fixture.

File: tests/test_resources/test_clusters/test_cluster.py (hunk 3 above)
Comment (BelSasha) [reply tests/test_resources/test_clusters/test_cluster.py:767]: to make sure we could assert that `'HTTP Request: POST {api_server_url}/resource/{cluster_uri}/cluster/status "HTTP/1.1 200 OK"' in cluster_logs` (line 774)

prob will change soon tho, if we'll print this log in a debug mode.

**Pull request merged**